scripts/modelbench/launch.sh

```diff
@@ -1,62 +1,51 @@
-TAG=${1:-ww09.2}
-PRECISION=${2:-float32}
-TEST_MODE=${3:-inference}
-TEST_SHAPE=${4:-static}
+#!/bin/bash
+set -xe
 
-TORCH_REPO=${5:-https://github.com/pytorch/pytorch.git}
-TORCH_BRANCH=${6:-nightly}
-TORCH_COMMIT=${7:-nightly}
-
-DYNAMO_BENCH=${8:-fea73cb}
-
-AUDIO=${9:-0a652f5}
-TEXT=${10:-c4ad5dd}
-VISION=${11:-f2009ab}
-DATA=${12:-5cb3e6d}
-TORCH_BENCH=${13:-a0848e19}
-
-THREADS=${14:-all}
-CHANNELS=${15:-first}
-WRAPPER=${16:-default}
-HF_TOKEN=${17:-hf_xx}
-BACKEND=${18:-inductor}
-SUITE=${19:-all}
-MODEL=${20:-resnet50}
-TORCH_START_COMMIT=${21:-${TORCH_BRANCH}}
-TORCH_END_COMMIT=${22:-${TORCH_START_COMMIT}}
-SCENARIO=${23:-accuracy}
-KIND=${24:-crash} # issue kind crash/drop
-PERF_RATIO=${25:-1.1}
-EXTRA=${26}
-
-echo "TAG" : $TAG
-echo "PRECISION" : $PRECISION
-echo "TEST_MODE" : $TEST_MODE
-echo "TEST_SHAPE" : $TEST_SHAPE
-echo "TORCH_REPO" : $TORCH_REPO
-echo "TORCH_BRANCH" : $TORCH_BRANCH
-echo "TORCH_COMMIT" : $TORCH_COMMIT
-echo "DYNAMO_BENCH" : $DYNAMO_BENCH
-echo "AUDIO" : $AUDIO
-echo "TEXT" : $TEXT
-echo "VISION" : $VISION
-echo "DATA" : $DATA
-echo "TORCH_BENCH" : $TORCH_BENCH
-echo "THREADS" : $THREADS
-echo "CHANNELS" : $CHANNELS
-echo "WRAPPER" : $WRAPPER
-echo "BACKEND" : $BACKEND
-echo "SUITE" : $SUITE
-echo "MODEL" : $MODEL
-echo "TORCH_START_COMMIT" : $TORCH_START_COMMIT
-echo "TORCH_END_COMMIT" : $TORCH_END_COMMIT
-echo "SCENARIO" : $SCENARIO
-echo "KIND" : $KIND
+# set params firstly
+# default
+TAG="ww18.4"
+PRECISION="float32"
+TEST_MODE="inference"
+SHAPE="static"
+TORCH_REPO="https://github.com/pytorch/pytorch.git"
+TORCH_COMMIT="nightly"
+DYNAMO_BENCH="fea73cb"
+AUDIO="0a652f5"
+TEXT="c4ad5dd"
+VISION="f2009ab"
+DATA="5cb3e6d"
+TORCH_BENCH="a0848e19"
+THREADS="all"
+CHANNELS="first"
+WRAPPER="default"
+HF_TOKEN=""
+BACKEND="inductor"
+SUITE="all"
+MODEL="resnet50"
+TORCH_START_COMMIT="nightly"
+TORCH_END_COMMIT="${TORCH_START_COMMIT}"
+SCENARIO="accuracy"
+KIND="crash" # issue kind crash/drop
+PERF_RATIO="-1.1"
+EXTRA=""
+# get value from param
+if [[ "$@" != "" ]];then
+    echo "" tmp.env
+    for var in "$@"
+    do
+        if [[ "${var}" == "EXTRA="* ]];then
+            EXTRA="${@/*EXTRA=}"
+            break
+        else
+            echo "$var" >> tmp.env
+        fi
+        shift
+    done
+    source tmp.env && rm -rf tmp.env
+fi
```

Comment:
This optimize method looks great! Just have 1 small concern about the nested source tmp.env, for example, the entrance.sh will source the itself temp.env, and then the entrance.sh call the launch.sh, and the launch.sh will also source itself temp.env again, will it has potential issue?

Reply:
No, before/after source will clean the tmp.env. they are from different env file exactly, and env context is from itself params. 
Fix an issue 'echo "" > tmp.env'